source/common/http/header_utility.cc

```diff
@@ -227,6 +227,29 @@ bool HeaderUtility::isEnvoyInternalRequest(const RequestHeaderMap& headers) {
          internal_request_header->value() == Headers::get().EnvoyInternalRequestValues.True;
 }
 
+void HeaderUtility::stripTrailingHostDot(RequestHeaderMap& headers) {
+  auto host = headers.getHostValue();
+  // Find last dot and return if not found or take off end if last
+  auto dot_index = host.rfind('.');
+  if (dot_index == std::string::npos) {
+    return;
+  } else if (dot_index == (host.size() - 1)) {
+    host.remove_suffix(1);
+    headers.setHost(host);
+    return;
+  }
+  // Check if the dot is just before a colon, which means it must be the port
```

Comment:
`If the dot is just before a colon, it must be preceding the port number.`

Reply:
Will change.